ADVANCED.md

```diff
@@ -0,0 +1,207 @@
+# BigInt: Arbitrary precision integers in JavaScript
+
+Daniel Ehrenberg, Igalia. Stage 3
+
+Thanks for help and feedback on this effort from Brendan Eich, Waldemar Horwat, Jaro Sevcik, Benedikt Meurer, Michael Saboff, Adam Klein and others.
+
+## Code samples:
+
+Find the nth prime:
+
+```js
+// Takes a BigInt as an argument and returns a BigInt
+function nthPrime(nth) {
+  function isPrime(p) {
+    for (let i = 2n; i < p; i++) {
+      if (p % i === 0n) return false;
+    }
+    return true;
+  }
+  for (let i = 2n; ; i++) {
+    if (isPrime(i)) {
+      if (--nth === 0n) return i;
+    }
+  }
+}
+```
+
+Read and add two unsigned 64-bit integers in asm.js on the heap:
+
+```js
+function Add64Module(stdlib, foreign, buffer) {
+  "use asm";
+  var cast = stdlib.BigInt.asUintN;
+  var values = new stdlib.BigUint64Array(buffer);
+  function add64(aIndex, bIndex) {
+    aIndex = aIndex|0;
+    bIndex = bIndex|0;
+    var aValue = values[aIndex>>3];
+    var bValue = values[bIndex>>3];
+    return cast(64, aValue + bValue);
+  }
+  return { add64: add64 };
+}
+```
```

Comment:
I've removed this example from the main README, as it reflects an old plan that will not happen.